presto-main/src/main/java/com/facebook/presto/FullConnectorSession.java

```diff
@@ -109,6 +114,12 @@ public long getStartTime()
     }
 
     @Override
+    public <T> T getSystemProperty(String name, Class<T> type)
+    {
+       return type.cast(systemProperties.get(name));
```

Comment:
Obviously, we will investigate more. Right now I am not quite sure how to do it. I will let is stay as it is for now. We will definitely iterate over this as we get more feedback on the RE2J support.